pkgs/development/libraries/libwebsockets/default.nix

```diff
@@ -11,7 +11,8 @@ stdenv.mkDerivation rec {
     sha256 = "1hv2b5r6sg42xnqhm4ysjvyiz3cqpfmwaqm33vpbx0k7arj4ixvy";
   };
 
-  buildInputs = [ cmake openssl zlib libuv ];
+  buildInputs = [ cmake openssl zlib ];
+  propagatedBuildInputs = [ libuv ];
```

Comment:
maybe ```openssl``` too

Reply:
Is it required for compiling (in example used in public header)?

Reply:
yes, but surrounded with #ifdefs 
https://github.com/warmcat/libwebsockets/blob/master/lib/libwebsockets.h#L195

Reply:
```libuv``` is not a strict dependency as well, a simple websocket servers could be created without linking with ```libuv```.

Reply:
yeah, maybe I should revert this.

Reply:
c1de354a3db238ddd1c125a77fd931566b2dda47